docs/concepts/admission-controller.md

```diff
@@ -12,4 +12,42 @@ This handler checks if the incoming request contains a Kubernetes secret with a
 
 Namespaces are the backing entities of Gardener projects in which shoot clusters objects reside.
 This validation handler protects active namespaces against premature deletion requests.
-Therefore, it denies deletion requests if a namespace still contains shoot clusters or if it belongs to a non-deleting Gardener project (w/o `.metadata.deletionTimestamp`).
\ No newline at end of file
+Therefore, it denies deletion requests if a namespace still contains shoot clusters or if it belongs to a non-deleting Gardener project (w/o `.metadata.deletionTimestamp`).
+
+### Resource Size Validator
+
+Since users directly apply Kubernetes native objects to the Garden cluster, it also involves the risk of being vulnerable to DoS attacks because these resources are read by controllers in the Gardener landscape.
+One example is the creation of shoot clusters with large annotation values (up to 256 kB per value) which can cause severe out-of-memory issues for the Gardenlet component.
+[Vertical autoscaling](https://github.com/kubernetes/autoscaler/tree/master/vertical-pod-autoscaler) can help to mitigate such situations, but we cannot expect to scale infinitely and thus need means to block the attack itself.
+
+The Resource Size Validator checks arbitrary incoming admission requests against a configured maximum size for the resource's group-version-kind combination and denies the request if the contained object exceeds the quota.
+
+Example for Gardener Admission Controller configuration:
+```yaml
+server:
+  resourceAdmissionConfiguration:
+    limits:
+    - apiGroups: ["core.gardener.cloud"]
+      apiVersions: ["*"]
+      resources: ["shoots", "plants"]
+      size: 100k
+    - apiGroups: [""]
+      apiVersions: ["v1"]
+      resources: ["secrets"]
+      size: 100k
+    unrestrictedUsers:
```

Comment:
I'm wondering whether it makes sense to use the `[]rbacv1.Subject` type here. WDYT?

Reply:
Yes, good idea. I switched to it in 87272f1165ddfde470844ce7934a026bf6af1033.